src/governance/LendingTermOnboarding.sol

```diff
@@ -0,0 +1,142 @@
+/// where CREDIT holders can veto the onboarding.
+/// Only terms that have been deployed through this factory can be onboarded.
+/// A term can be onboarded for the first time, or re-onboarded after it has been offboarded.
+contract LendingTermOnboarding is VoltGovernor {
+
+    uint256 MIN_DELAY_BETWEEN_PROPOSALS = 7 days;
+    address public immutable lendingTermImplementation;
+    address public immutable guildToken;
+    uint256 public immutable gaugeType;
+    mapping(address=>uint256) public created;
+    mapping(address=>uint256) public lastProposal;
+    
+    constructor(
+        address _lendingTermImplementation,
+        address _guildToken,
+        uint256 _gaugeType,
+        address _core,
+        address _timelock,
+        address _token,
+        uint256 initialVotingDelay,
+        uint256 initialVotingPeriod,
+        uint256 initialProposalThreshold,
+        uint256 initialQuorum
+    )
+        VoltGovernor(
+            _core,
+            _timelock,
+            _token,
+            initialVotingDelay,
+            initialVotingPeriod,
+            initialProposalThreshold,
+            initialQuorum
+        )
+    {
+        lendingTermImplementation = _lendingTermImplementation;
+        guildToken = _guildToken;
+        gaugeType = _gaugeType;
+    }
+
+    /// @notice Create a new LendingTerm and initialize it.
+    function createTerm(LendingTerm.LendingTermParams calldata params) external returns (address) {
+        address term = Clones.clone(lendingTermImplementation);
+        LendingTerm(term).initialize(
+            address(LendingTerm(lendingTermImplementation).core()),
+            LendingTerm(lendingTermImplementation).getReferences(),
+            params
+        );
+        created[term] = block.timestamp;
+        return term;
+    }
+
+    /// @dev override to prevent arbitrary calls to be proposed
+    function propose(
+        address[] memory /* targets*/,
+        uint256[] memory /* values*/,
+        bytes[] memory /* calldatas*/,
+        string memory /* description*/
+    ) public pure override(IGovernor, Governor) returns (uint256) {
+        revert("LendingTermOnboarding: cannot propose arbitrary actions");
+    }
+
+    /// @notice Propose the onboarding of a term
+    function proposeOnboard(address term) external returns (uint256 proposalId) {
+        // Check that the term has been created by this factory
+        require(created[term] != 0, "LendingTermOnboarding: invalid term");
+
+        // Check that the term was not subject to an onboard vote recently
+        require(lastProposal[term] + MIN_DELAY_BETWEEN_PROPOSALS < block.timestamp, "LendingTermOnboarding: recently proposed");
+        lastProposal[term] = block.timestamp;
+        
+        // Check that the term is not already active
+        bool isGauge = GuildToken(guildToken).isGauge(term);
+        require(!isGauge, "LendingTermOnboarding: active term");
```

Comment:
if gauge was active, then was offboarded or deprecated, it can be added again

Reply:
yes that's the goal, certain terms for instance "Lend 1600 CREDIT per 1 rETH of collateral" can be bad at one point (and should get offboarded) but become good again later when market conditions change